### PR TITLE
feat: add redshift extractor

### PR DESCRIPTION
**Pull request by scortier**
## Deliverables

- [x] add unit tests (_TODO:Check with real cluster data from redshift_)
- [x] add extractor
- [x] add README.md in ```plugins/extractors/redshift```, defining output
- [x] register your extractor ```plugins/extractors/populate.go```
- [x] add extractor the extractor list in ```docs/reference/extractor.md```

### Table
| Field | Sample Value |
| :---- | :---- |
| `urn` | `my_database.my_table` |
| `name` | `my_table` |
| `source` | `redshift` |
| `schema` | [][Column] |

### Column
| Field | Sample Value |
| :---- | :---- |
| `name` | `total_price` |
| `description` | `item's total price` |
| `data_type` | `decimal` |
| `is_nullable` | `true` |
| `length` | `1243` |

#### Issue #88

### CODE REVIEW

File: plugins/extractors/redshift/redshift.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	"github.com/odpf/meteor/registry"
+	"github.com/odpf/meteor/utils"
+	"github.com/odpf/salt/log"
+	"strings"
+)
+
+//go:embed README.md
+var summary string
+
+var defaultExcludes = []string{"information_schema", "pg_catalog", "pg_internal", "public"}
```
Comment (mabdh) [inline plugins/extractors/redshift/redshift.go:24]: these exclude list is postgres specific, I am not sure it fits within redshift context

File: plugins/extractors/redshift/redshift.go [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+// Config holds the set of configuration for the metabase extractor
+type Config struct {
+	ClusterID string `mapstructure:"cluster_id" validate:"required"`
+	DbName    string `mapstructure:"db_name" validate:"required"`
```
Comment (mabdh) [inline plugins/extractors/redshift/redshift.go:29]:
```suggestion
	DBName    string `mapstructure:"db_name" validate:"required"`
```

File: plugins/extractors/redshift/redshift.go [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+	DbUser    string `mapstructure:"db_user" validate:"required"`
+	AwsRegion string `mapstructure:"aws_region" validate:"required"`
```
Comment (mabdh) [inline plugins/extractors/redshift/redshift.go:31]:
```suggestion
	AWSRegion string `mapstructure:"aws_region" validate:"required"`
```

File: plugins/extractors/redshift/redshift.go [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+//go:embed README.md
+var summary string
+
+var defaultExcludes = []string{"information_schema", "pg_catalog", "pg_internal", "public"}
+
+// Config holds the set of configuration for the metabase extractor
+type Config struct {
+	ClusterID string `mapstructure:"cluster_id" validate:"required"`
+	DbName    string `mapstructure:"db_name" validate:"required"`
+	DbUser    string `mapstructure:"db_user" validate:"required"`
```
Comment (mabdh) [inline plugins/extractors/redshift/redshift.go:30]:
```suggestion
	DBUser    string `mapstructure:"db_user" validate:"required"`
```

File: plugins/extractors/redshift/redshift.go (hunk 1 above)
Comment (scortier) [reply plugins/extractors/redshift/redshift.go:24]: In db querying, redshift works in similar way as postgres,  [the list here is redshift specific](https://docs.aws.amazon.com/redshift/latest/mgmt/data-api.html) but yes till we test it with real cluster data we can remove it.

**Review by mabdh — APPROVED** [review]

**Pull request merged**